src/base/Roles/CrossChain/CrossChainTellerBase.sol

```diff
@@ -0,0 +1,204 @@
+     * @dev Callable by MULTISIG_ROLE.
+     */
+    function stopMessagesFromChain(uint32 chainSelector) external requiresAuth {
+        Chain storage chain = selectorToChains[chainSelector];
+        chain.allowMessagesFrom = false;
+
+        emit ChainStopMessagesFrom(chainSelector);
+    }
+
+    /**
+     * @notice Stop messages to a chain.
+     * @dev Callable by MULTISIG_ROLE.
+     */
+    function stopMessagesToChain(uint32 chainSelector) external requiresAuth {
+        Chain storage chain = selectorToChains[chainSelector];
+        chain.allowMessagesTo = false;
+
+        emit ChainStopMessagesTo(chainSelector);
+    }
+
+    /**
+     * @notice Set the gas limit for messages to a chain.
+     * @dev Callable by OWNER_ROLE.
+     */
+    function setChainGasLimit(uint32 chainSelector, uint64 messageGasLimit) external requiresAuth {
+        if (messageGasLimit == 0) {
+            revert CrossChainLayerZeroTellerWithMultiAssetSupport_ZeroMessageGasLimit();
+        }
+        Chain storage chain = selectorToChains[chainSelector];
+        chain.messageGasLimit = messageGasLimit;
+
+        emit ChainSetGasLimit(chainSelector, messageGasLimit);
+    }
+
+    
+    /**
+     * @dev function to deposit into the vault AND bridge cosschain in 1 call
+     * @param depositAsset ERC20 to deposit
+     * @param depositAmount amount of deposit asset to deposit
+     * @param minimumMint minimum required shares to receive
+     * @param data Bridge Data
+     */
+    function depositAndBridge(ERC20 depositAsset, uint256 depositAmount, uint256 minimumMint, BridgeData calldata data) external payable{
+        uint shareAmount = _erc20Deposit(depositAsset, depositAmount, minimumMint, msg.sender);
+        bridge(shareAmount, data);
+    }
+
+
+    /**
+     * @notice Preview fee required to bridge shares in a given feeToken.
+     */
+    function previewFee(uint256 shareAmount, BridgeData calldata data)
+        external
+        view
+        returns (uint256 fee)
+    {
+        return _quote(shareAmount, data);
+    }
+
+    /**
+     * @dev bridging code to be done without deposit, for users who already have vault tokens
+     * @param shareAmount to bridge
+     * @param data bridge data
+     */
+    function bridge(uint256 shareAmount, BridgeData calldata data) public payable returns(bytes32 messageId){
+        _beforeBridge(shareAmount, data);
+        messageId = _bridge(shareAmount, data);
+        _afterBridge(shareAmount, data, messageId);
+    }
+
+    /**
+     * @dev code to run before bridging, includes checks and a burn of shares
+     * @dev in the CrossChainTellerWithGenericBridge implementation this is inspired by, some data processing is done beforehand,
+     * here that is not done in case other implementations need more flexibility. But I will check to see if it should be done here.
+     * @param shareAmount to burn
+     * @param data unused but potentially needed in an override
```

Comment:
did you intend for this _beforeBridge function to possibly be overridden in some future cases? If so, then I think we need a virtual

Reply:
It was virtual, but after talking with Jun, he thinks it should be locked off, and any override logic should go in _bridge(). 

I think what I'll do is make this private so it's more clear the intentions for it to just be organizational as a hook